Fix ConvertToEdoc:getRealPath() - Incorrect Real Path Check Causing Asset Removal

D3V41 opened the pull request:
### Status Quo
- The `getRealPath()` method does not correctly validate the `uri` parameter and is returning incorrect results.

![image](https://github.com/user-attachments/assets/68918b81-716d-4a47-82e3-a70da16715b5)


### Changes
- Fixed the `getRealPath()` method based on the values of `ConvertToEdoc.realPath` and `ConvertToEdoc.contextPath`.
- Current values:
    - `ConvertToEdoc.realPath`: `/home/vagrant/tomcat/apache-tomcat-9.0.41/webapps/oscar/`
    - `ConvertToEdoc.contextPath` : `/oscar`

- Based on these values, it appears that `getRealPath()` is intended to resolve the real path of the given `uri` within the `oscar` project directory. The implementation has been updated accordingly.

warrendennis reviewed (changes requested):
More information needed. Please demonstrate where and how this fails, and how to test if it's fixed.

The intention of this method is to filter invalid relative paths from being executed in eForms.  If a relative URI does not have a valid "real path" in the system, it will not be used See comment on line 517:
	/*
			 * NO EXTERNAL LINKS. These are removed.
			 * eForms are often imported from unknown sources.
			 * Developers tend to use insecure CDN's, links to images, tracking tokens,
			 * and advertisements.
			 */

D3V41 commented:
Code:
```java
/*
* NO EXTERNAL LINKS. These are removed.
* eForms are often imported from unknown sources.
* Developers tend to use insecure CDN's, links to images, tracking tokens,
* and advertisements.
*/
if(path.startsWith("http") || path.startsWith("HTTP")) {
   element.remove();
}

// internal GET links are validated.
else if( path.contains("?") ) {
   // image or link paths with parameters
   parameters = path.split("\\?")[1];
}

else if(! path.isEmpty()) {
   // these are most likely relative context paths
   path = getRealPath( path );
   if(! path.isEmpty()) {
	potentialFilePaths.add(path);
   }
}
``` 

There are three main conditions in the above code:

1. **External resource check** – It removes any external links (e.g., starting with http or HTTP).
2. **Links with query parameters** – The `parameters` are extracted for further validation within `OscarDocument`.
3. **Relative paths** – Checks whether the resource is a relative context path (as indicated by `// these are most likely relative context paths`) and verifies if it is present in the `/oscar` directory.

✅ I’ve fixed the logic related to this third condition (`getRealPath()`).

### To reproduce the issue on the `main` branch:

1. Create an `Annual` form in the EChart
   - <img width="239" height="116" alt="1" src="https://github.com/user-attachments/assets/f3b6c641-e470-47fe-b02f-83d70bd7feb0" />

2. Attach the `Annual` form to a Consultation Request, then click `Submit Consultation Request & Print Preview`.
   - <img width="294" height="113" alt="2" src="https://github.com/user-attachments/assets/64762dc0-4e5a-45ed-946f-c490a01c5ae2" />
   - <img width="475" height="384" alt="3" src="https://github.com/user-attachments/assets/8371a5a0-e5fa-413e-8a0f-92b90c996d30" />

3. Upload the following sample eForm, but replace the `img` path with a any valid image from the Image Library first:
   - <img width="866" height="269" alt="3 1" src="https://github.com/user-attachments/assets/bf4ed593-b602-43f5-8002-1e448632ed1f" />

   - ```html
      <!DOCTYPE html>
      <html lang="en">
      <head>
          <title>Form with Image</title>
      </head>
      <body>
          <form action="#" method="post">
              <script src="${oscar_javascript_path}jquery/jquery-3.6.4.min.js"></script>
              <img src="${oscar_image_path}labreq3.png" alt="Sample Image">
          </form>
      </body>
      </html>
      ``` 

4. Open the uploaded eForm in the EChart and click `Save`.
   - <img width="162" height="95" alt="4" src="https://github.com/user-attachments/assets/db571c1e-2f97-4c23-baf6-52291450d98e" />

5. Query the `eform_data` table and observe the HTML. The `<script>` tag will have been stripped from the content:
   - ```
      SELECT * FROM eform_data ed ORDER BY ed.fdid DESC;
      ``` 

Now it is fixed.

warrendennis commented:
Not sure I follow. 
Is the problem that the <script> tag is stripped from the eForm data?  How is this related to a relative path not being resolved correctly?
